src/Playwright/Core/Artifact.cs

```diff
@@ -69,11 +69,13 @@ public async Task SaveAsAsync(string path)
                 return;
             }
             System.IO.Directory.CreateDirectory(Path.GetDirectoryName(path));
-            await using var stream = await _channel.SaveAsStreamAsync().ConfigureAwait(false);
-
-            using (var fileStream = new FileStream(path, FileMode.Create, FileAccess.Write))
+            var stream = await _channel.SaveAsStreamAsync().ConfigureAwait(false);
+            await using (stream.ConfigureAwait(false))
             {
-                await stream.StreamImpl.CopyToAsync(fileStream).ConfigureAwait(false);
+                using (var fileStream = new FileStream(path, FileMode.Create, FileAccess.Write))
```

Comment:
You should be able to put these two usings at the same level

```
await using (stream.ConfigureAwait(false))
using (var fileStream = new FileStream(path, FileMode.Create, FileAccess.Write))
{
    ...
}
```

Reply:
linting rules are conflicting there thats why I put it that way: https://github.com/dotnet/roslyn-analyzers/issues/5712